pkg/controllers/updaterun/execution.go

```diff
@@ -604,9 +604,13 @@ func checkClusterUpdateResult(
 	return false, nil
 }
 
-// buildApprovalRequestObject creates an approval request object for after-stage tasks.
+// buildApprovalRequestObject creates an approval request object for before-stage or after-stage tasks.
 // It returns a ClusterApprovalRequest if namespace is empty, otherwise returns an ApprovalRequest.
 func buildApprovalRequestObject(namespacedName types.NamespacedName, stageName, updateRunName string) placementv1beta1.ApprovalRequestObj {
+	stageTask := "before"
+	if strings.Contains(namespacedName.Name, "after") {
+		stageTask = "after"
+	}
```

Comment:
so the label value actually comes from the name of the task? It's bit weird that we have a label whose accuracy depends on the naming of the object. If so, the naming itself already contains enough information and it's more visible.